src/newgrf_gui.cpp

```diff
@@ -1525,10 +1528,10 @@ struct NewGRFWindow : public Window, NewGRFScanCallback {
 		*entry = c;
 
 		/* Select next (or previous, if last one) item in the list. */
-		int new_pos = this->avail_pos + 1;
-		if (new_pos >= (int)this->avails.size()) new_pos = this->avail_pos - 1;
+		uint new_pos = this->avail_pos + 1;
+		if (new_pos >= this->avails.size()) new_pos = this->avail_pos - 1;
 		this->avail_pos = new_pos;
-		if (new_pos >= 0) this->avail_sel = this->avails[new_pos];
+		this->avail_sel = this->avails[new_pos];
```

Comment:
I think a safety check is still needed, `new_pos` can be `UINT_MAX` if `this->avail_pos` was 0 and only 1 element in `this->avails`